vendor/src/github.com/docker/engine-api/client/interface.go

```diff
@@ -100,11 +100,11 @@ type NodeAPIClient interface {
 
 // ServiceAPIClient defines API client methods for the services
 type ServiceAPIClient interface {
-	ServiceCreate(ctx context.Context, service swarm.ServiceSpec) (types.ServiceCreateResponse, error)
+	ServiceCreate(ctx context.Context, service swarm.ServiceSpec, headers map[string][]string) (types.ServiceCreateResponse, error)
```

Comment:
Was there an engine-api review? Having `headers` here is the wrong level of abstraction. This should probably be an option struct.